src/main/ApplicationImpl.cpp

```diff
@@ -1197,6 +1203,14 @@ ApplicationImpl::getMetrics()
     return *mMetrics;
 }
 
+bool
+ApplicationImpl::threadIsType(ThreadType type) const
```

Comment:
I don't think this should assert. For example, we currently don't always initialize a LEDGER_CLOSE thread since it's behind an experimental flag, but something like `releaseAssert(!isLedgerCloseThread)` should probably be valid whether the flag is set or not.

Reply:
Oh, yeah, this is also true!

Reply:
That's not what this is asserting. Your example would be fine here. The assertion is checking that the calling thread's ID is in the map. This ensures that the map has been properly constructed. It wouldn't be possible for a nonexistent thread to trigger this.

That is to say, this is a map from thread IDs to thread types, not thread types to thread ids. It should contain all running thread IDs. if it's missing one, then that's a bug.

Am I missing something?

Reply:
Shoot, ya you're right, sorry I flipped this key-value in my head for some reason/